pkg/controller/autoimport/apiserver_syncer.go

```diff
@@ -0,0 +1,113 @@
+package autoimport
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/stolostron/managedcluster-import-controller/pkg/constants"
+	"github.com/stolostron/managedcluster-import-controller/pkg/helpers"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	kevents "k8s.io/client-go/tools/events"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type apiServerSyncer struct {
+	client       client.Client
+	recorder     events.Recorder
+	mcRecorder   kevents.EventRecorder
+	importHelper *helpers.ImportHelper
+}
+
+func (s *apiServerSyncer) sync(ctx context.Context,
+	managedCluster *clusterv1.ManagedCluster, autoImportSecret *corev1.Secret) (reconcile.Result, error) {
+
+	apiServerURL, err := s.getAPIServerURL(autoImportSecret)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if apiServerURL == "" {
+		return reconcile.Result{}, nil
+	}
+
+	apiServerURL = strings.TrimSuffix(apiServerURL, "/")
+
+	clusterCopy := managedCluster.DeepCopy()
+	for _, config := range clusterCopy.Spec.ManagedClusterClientConfigs {
+		if config.URL == apiServerURL {
+			return reconcile.Result{}, nil
+		}
+	}
+
+	clusterCopy.Spec.ManagedClusterClientConfigs = append(clusterCopy.Spec.ManagedClusterClientConfigs,
+		clusterv1.ClientConfig{
+			URL: apiServerURL,
+		})
+
```

Comment:
do we need to add a log to indicate that the config URL in the spec is added by import controller?

Reply:
done